includes/promoted-jobs/class-wp-job-manager-promoted-jobs-status-handler.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * File containing the class WP_Job_Manager_Promoted_Jobs_Status_Handler.
+ *
+ * @package wp-job-manager
+ */
+
+/**
+ * Handles functionality related to the Promoted Jobs Status Update.
+ *
+ * @since $$next-version$$
+ */
+class WP_Job_Manager_Promoted_Jobs_Status_Handler {
+
+	/**
+	 * The name of the cron hook for updating the promoted job status.
+	 */
+	const CRON_HOOK = 'job_manager_promoted_jobs_status_update';
+
+	/**
+	 * The name of the option that stores the last time the cron job was executed.
+	 */
+	const OPTION_KEY = self::CRON_HOOK . '_last_execution';
+
+	/**
+	 * The frequency at which the cron job should be executed.
+	 */
+	private const FREQUENCY_UPDATE = HOUR_IN_SECONDS;
```

Comment:
I'm thinking it it would make more sense to use 5 minutes here in order to match the lower interval we have [here](https://github.com/Automattic/wpjobmanager.com/pull/561/files#diff-0a086bb7ed1691365420085e907d905bb7fdb67319650a46ef08bbcb44ff72faR59). WDYT?

Otherwise, if we have 2 promotions in a row, we would see the status update for the second one only after the cron job running `twicedaily`, right?

Reply:
@renatho Good point. As discussed on Slack, I pushed the commits 6b315a5, ee1fa71 and 8169dab to only run `fetch_updates()` if the user had redirected to the Promote Job form at least once (by clicking on the button in the modal to promote a job).

I also created issue #2558 to improve this logic.

Reply:
I'd suggest only change the name to something like "used promoted jobs" or "touched promoted jobs" instead of "has promoted jobs". Otherwise, it could be a little confusing since it doesn't represent exactly having a promoted job. WDYT?

Reply:
Sure! Renamed in ec759b8, and adjusted frequency update (which I also renamed to something friendlier) in c9fb860. Thanks!